contrib/fine-tuning/fine-tune-mlx.sh

```diff
@@ -38,11 +41,19 @@ node validator.js ${DATASET_PATH}
 
 # This step always pulls the latest base model from HF. Need to think about versioning and checksum to prevent model injection attacks
 echo "Test base model with the prompt 'Tell me about cdxgen'. Usually yields a low-quality response."
-mlx_lm.generate --model ${BASE_MODEL} --prompt "Tell me about cdxgen" --temp 0.05 --max-tokens 32000
+mlx_lm.generate --model ${BASE_MODEL} --prompt "Tell me about cdxgen" --temp 0.7 --max-tokens 16000
 
 # We use LoRA fine-tuning over DoRA due to better compatibility with vLLM and llama.cpp
-echo "Low-Rank Adaptation (LoRA) fine-tuning ${BASE_MODEL} with cdx1 dataset. This might take a while ..."
-mlx_lm.lora --model ${BASE_MODEL} --train --data ${DATASET_PATH} --adapter-path ${ADAPTERS_PATH} --mask-prompt --fine-tune-type lora --batch-size 1 --num-layers ${NUM_LAYERS} --iters 2000 --grad-checkpoint --max-seq-length 16000 --learning-rate "3e-5" --optimizer adam
+if [ "$TOOL_BASE_MODEL" = "cdx1-mini" ]; then
+  echo "Full fine-tune with cdx-docs dataset. This might take a while ..."
+  mlx_lm.lora --model ${BASE_MODEL} --train --data ${DATASET_PATH} --adapter-path ${ADAPTERS_PATH} --mask-prompt --fine-tune-type full --batch-size 2 --iters 1500 --grad-checkpoint --max-seq-length 8192 --learning-rate "1e-5" --optimizer adamw --num-layers 32 --seed 42
```

Comment:
For mini, we take a small 4B model and perform a full fine-tuning rather than lora/qlora.